Subject: [PATCH] rtkmipsel: rtl8197f: Tenda AC8 v1 machine instead of the HH71VM

Build mach-tenda-ac8.c (LEDs, reset button) as the only machine of the target,
and start the early console at 115200, the speed of the AC8 console.

---
diff --git a/target/linux/rtkmipsel/files/arch/mips/rtl8197f/Kconfig b/target/linux/rtkmipsel/files/arch/mips/rtl8197f/Kconfig
index 8b61097..3950e4c 100644
--- a/target/linux/rtkmipsel/files/arch/mips/rtl8197f/Kconfig
+++ b/target/linux/rtkmipsel/files/arch/mips/rtl8197f/Kconfig
@@ -84,12 +84,12 @@ config RTK_DUALIMAGE_FLASH_OFFSET
 
 menu "Realtek RTL8197F machine selection"
 
-config RTL8197_MACH_HH71VM
-	bool "Alcatel LINKHUB HH71VM"
+config RTL8197_MACH_TENDA_AC8
+	bool "Tenda AC8 v1"
 	default y
 	help
 	  Say 'Y' here if you want your kernel to support the
-	  Alcatel LINKHUB HH71VM (RTL8197FS + RTL8812FE).
+	  Tenda AC8 v1 (RTL8197FH-VG + RTL8812FR + RTL8367RB-VB).
 
 endmenu
 
diff --git a/target/linux/rtkmipsel/files/arch/mips/rtl8197f/Makefile b/target/linux/rtkmipsel/files/arch/mips/rtl8197f/Makefile
index 6e437b6..6fee8fd 100644
--- a/target/linux/rtkmipsel/files/arch/mips/rtl8197f/Makefile
+++ b/target/linux/rtkmipsel/files/arch/mips/rtl8197f/Makefile
@@ -30,4 +30,4 @@ obj-$(CONFIG_I2C_DESIGNWARE_PLATFORM)  += i2c.o
 #
 # Machines
 #
-obj-$(CONFIG_RTL8197_MACH_HH71VM)	+= mach-hh71vm.o
+obj-$(CONFIG_RTL8197_MACH_TENDA_AC8)	+= mach-tenda-ac8.o
diff --git a/target/linux/rtkmipsel/files/arch/mips/rtl8197f/prom.c b/target/linux/rtkmipsel/files/arch/mips/rtl8197f/prom.c
index 839ac76..1511545 100644
--- a/target/linux/rtkmipsel/files/arch/mips/rtl8197f/prom.c
+++ b/target/linux/rtkmipsel/files/arch/mips/rtl8197f/prom.c
@@ -85,6 +85,12 @@ void prom_putchar(char c)
 	return;
 }
 
+/* Tenda AC8: the console runs at 115200 from the first line on. */
+#ifdef CONFIG_RTL8197_MACH_TENDA_AC8
+#undef BSP_UART0_BAUD_DIVISOR
+#define BSP_UART0_BAUD_DIVISOR	((BSP_UART0_FREQ >> 4) / 115200)
+#endif
+
 static int bsp_serial_init(void)
 {
 	if (IS_8197F_VG()) {
